duckduckgo/duckduckgo-privacy-extension
# Surrogates submodule

**Reviewer:** @jdorweiler 

**CC:** @dharb 

## Description:
We moved surrogate files to a separate repo - https://github.com/duckduckgo/tracker-surrogates . This PR removes embedded surrogates and replaces them with surrogates from that repo (which is almost 1:1 change minus the different code style). We are pulling that other repo in through the magic of 🪄git submodules🦄 that this repo was alrady using for tests that Lucas added.

Notes:
- fb-sdk.js (FB click to play) was not pulled into surrogates because it's injected differently than other surrogates (not based on simple match rules in TR)
- I made changes to how `surrogates.txt` file is generated - it's now legacy and we only use it as a list of available surrogates, so I changed it so that it doesn't include full JS code now, only list of surrogate files
- We should remove the whole `surrogates.txt` but I attempted that and it's a time sink it touches multiple things in this repo AND privacy-grade repo

## Steps to test this PR:
1. Get on this branch
2. `git submodule update --init --recursive` to pull the submodule code
3. Build the extension
4. Go to http://privacy-test-pages.glitch.me/privacy-protections/surrogates/ all tests (but the second "integrity" one) should be green
5. Go to cnn.com, check in devtools (Chrome) that surrogates for gtp, outbrain, comscore, etc. are getting fetched, check in console (Chrome, Firefox) that global objects from those files are available and overwritten with our noops

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
